app/org/apache/spark/deploy/history/SparkDataCollection.scala

```diff
@@ -192,6 +195,46 @@ class SparkDataCollection(applicationEventListener: ApplicationEventListener,
     _executorData
   }
 
+  override def getExecutorTrackingData(): SparkExecutorData = {
+    if (_executorTrackingData == null) {
+      _executorTrackingData = new SparkExecutorData()
+
+      val applicationFinishTime = getGeneralData().getEndTime
+
+      executorsTrackingListener.executorIdToData.foreach { case (execId, lifeData) =>
+        val info = new ExecutorInfo()
+
+        // TODO: Combine block manager events with executor events because they are in same context
+        //val status = executorsListener.storageStatusList(statusId)
+
+        info.execId = execId
+        //info.execId = status.blockManagerId.executorId
+        //info.hostPort = status.blockManagerId.hostPort
+        //info.rddBlocks = status.numBlocks
+
+        // Use a customized listener to fetch the peak memory used, the data contained in status are
+        // the current used memory that is not useful in offline settings.
+        //info.memUsed = storageStatusTrackingListener.executorIdToMaxUsedMem.getOrElse(info.execId, 0L)
+        //info.maxMem = status.maxMem
+        //info.diskUsed = status.diskUsed
+        info.activeTasks = executorsListener.executorToTasksActive.getOrElse(info.execId, 0)
+        info.failedTasks = executorsListener.executorToTasksFailed.getOrElse(info.execId, 0)
+        info.completedTasks = executorsListener.executorToTasksComplete.getOrElse(info.execId, 0)
+        info.totalTasks = info.activeTasks + info.failedTasks + info.completedTasks
+        info.duration = executorsListener.executorToDuration.getOrElse(info.execId, 0L)
+        info.inputBytes = executorsListener.executorToInputBytes.getOrElse(info.execId, 0L)
+        info.shuffleRead = executorsListener.executorToShuffleRead.getOrElse(info.execId, 0L)
```

Comment:
where are you using shuffleread, inputBytes etc. only startTime and finishTime are used, right ? Do you plan to use the rest later sometime